include/libsemigroups/fpsemi-examples.hpp

```diff
@@ -0,0 +1,539 @@
+  std::vector<relation_type> stellar_monoid(size_t l);
+
+  //! Construct a presentation for the dual symmetric inverse monoid.
+  //!
+  //! Returns a vector of relations giving a semigroup presentation defining the
+  //! dual symmetric inverse monoid of degree `n`. The argument `val` determines
+  //! the specific presentation which is returned. The options are:
+  //! * `author::Easdown + author::East + author::FitzGerald` (from Section 3 of
+  //! [10.48550/arxiv.0707.2439](https://doi.org/10.48550/arxiv.0707.2439))
+  //!
+  //! \param n the degree
+  //! \param val the author of the presentation
+  //!
+  //! \returns A `std::vector<relation_type>`
+  //!
+  //! \throws LibsemigroupsException if `n < 3`, or the author value is not
+  //! `author::Easdown + author::East + author::FitzGerald`.
+  std::vector<relation_type> dual_symmetric_inverse_monoid(size_t n,
+                                                           author val);
+
+  //! Construct a presentation for the uniform block bijection monoid.
+  //!
+  //! Returns a vector of relations giving a semigroup presentation defining the
+  //! uniform block bijection monoid of degree `n`. The argument `val`
+  //! determines the specific presentation which is returned. The only option
+  //! is:
+  //! * `author::FitzGerald` (see
+  //! [10.1017/s0004972700037692](https://doi.org/10.1017/s0004972700037692))
+  //!
+  //! \param n the degree
+  //! \param val the author of the presentation
+  //!
+  //! \returns A `std::vector<relation_type>`
+  //!
+  //! \throws LibsemigroupsException if `n < 3`, or the author value is not
+  //! `author::FitzGerald`.
+  std::vector<relation_type> uniform_block_bijection_monoid(size_t n,
+                                                            author val);
+
+  //! Construct a presentation for the partition monoid.
+  //!
+  //! Returns a vector of relations giving a semigroup presentation defining the
+  //! partition monoid of degree `n`. The argument `val` determines the specific
+  //! presentation which is returned. The options are:
+  //! * `author::Machine`
+  //! * `author::East` (see Theorem 41 of
+  //! [10.1016/j.jalgebra.2011.04.008](https://doi.org/10.1016/j.jalgebra.2011.04.008))
+  //!
+  //! \param n the degree
+  //! \param val the author of the presentation
+  //!
+  //! \returns A `std::vector<relation_type>`
+  //!
+  //! \throws LibsemigroupsException if (`val = author::Machine` and `n != 3`),
+  //! or `(val = author::East` and `n < 4`).
+  std::vector<relation_type> partition_monoid(size_t n, author val);
+
+  //! Construct a presentation for the singular part of the Brauer monoid.
+  //!
+  //! Returns a vector of relations giving a semigroup presentation for the
+  //! singular part of the Brauer monoid of degree `n`, as in Theorem 5 of
+  //! [10.21136/MB.2007.134125](https://doi.org/10.21136/MB.2007.134125).
+  //!
+  //! \param n the degree
+  //!
+  //! \returns A `std::vector<relation_type>`
+  //!
+  //! \noexcept
```

Comment:
Really?

Reply:
Are you sure this function returns the right answer when `n` is `0` for example?